applications/nrf_desktop/src/modules/Kconfig.usb_state

```diff
@@ -206,20 +175,29 @@ config USBD_HID_IN_BUF_COUNT
 config DESKTOP_USB_STACK_NEXT_DISABLE_ON_VBUS_REMOVAL
 	bool
 	default y if SOC_SERIES_NRF54HX
-	depends on DESKTOP_USB_INIT_THREAD
-	select REBOOT
-	help
-	  Disable USB stack on VBUS removal. This is a workaround for the USB
-	  driver not working correctly on the nRF54HX SoC. After the USB cable
-	  is removed, the USB driver is powered down and doesn't call appropriate
-	  callbacks to the USB stack. It may lead to the USB stack ending in a
-	  broken state. Calling usbd_disable() on VBUS removal workarounds this
-	  issue. When this option is enabled, the usb_init_thread will be used to
-	  call the usbd_enable() and usbd_disable() functions on VBUS events. This
-	  is done because the usbd_enable() function blocks if the USB cable is
-	  not connected. A separate thread allows the application to trigger a
-	  reboot from the workqueue context to prevent blocking the application
-	  forever.
+	help
+	  Disable the USB stack on VBUS removal and enable it on VBUS ready to
+	  workaround UDC driver limitations of the nRF54HX SoC.
+
+	  After the USB cable is removed, the USB driver is powered down and
+	  doesn't call appropriate callbacks to the USB stack. It may lead to
+	  the USB stack ending in a broken state. Calling usbd_disable() on VBUS
+	  removal workarounds this issue.
+
+	  The usbd_enable() function blocks until USB cable is connected. If the
+	  cable is not connected during the predefined wait period
+	  (CONFIG_UDC_DWC2_USBHS_VBUS_READY_TIMEOUT), the function returns
+	  earlier with a timeout error. The function is called on VBUS ready to
+	  avoid polling the USB periodically.
```

Comment:
`... to avoid calling usb_enable() periodically` (USB polling in context of USB HID device may be incorrectly associated with polls of USB host)

Reply:
Done